datafusion/core/src/physical_optimizer/join_selection.rs

```diff
@@ -108,197 +105,49 @@ fn supports_collect_by_thresholds(
 }
 
 /// Predicate that checks whether the given join type supports input swapping.
+#[deprecated(since = "45.0.0", note = "use JoinType::supports_swap instead")]
```

Comment:
This code is moved elsewhere, I chose to mark it as deprecated to help my migration as well as forks that might use these functions (e.g. Synnadas)